python/ray/data/dataset.py

```diff
@@ -2619,6 +2625,21 @@ def write_datasource(
         finally:
             progress.close()
 
+    def iterator(self) -> DatasetIterator:
+        """Return a :class:`DatasetIterator <ray.data.DatasetIterator>` that
+        can be used to repeatedly iterate over the dataset.
+        Examples:
+            >>> import ray
+            >>> for batch in ray.data.range(
+            ...     1000000
+            ... ).iterator().iter_batches(): # doctest: +SKIP
+            ...     print(batch) # doctest: +SKIP
+
+        It is recommended to use `DatasetIterator` methods over directly
+        calling methods such as `iter_batches()`.
```

Comment:
Nit: Make it a note. Also, inlined code that aren't cross-reference links should be double-backticked.
```suggestion
        .. note::
            It is recommended to use ``DatasetIterator`` methods over directly
            calling methods such as ``iter_batches()``.
```